debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -122,20 +124,20 @@ protected void doInitialize() {
      */
     @Override
     protected void doStart() {
-        thread = new Thread(this::execute, "mysql-snapshot-" + context.serverName());
-        thread.start();
+        executorService = Threads.newSingleThreadExecutor(MySqlConnector.class, context.serverName(), "snapshot");
+        executorService.execute(this::execute);
     }
 
     @Override
     protected void doStop() {
         logger.debug("Stopping snapshot reader");
+        executorService.shutdown();
```

Comment:
It's a nice thought, but unfortunately this change also causes the thread to leak, if the snapshot reader is used via the chained reader. In that case, this method will only be invoked for the last reader in the chain (usually the binlog reader). I'll move it to `doCleanup()` and merge it. Going forward we should see how to simplify the lifecycle of these readers, it seems hard to comprehend.